src/main/java/org/openstreetmap/atlas/geography/atlas/complete/CompleteNode.java

```diff
@@ -189,6 +190,19 @@ public boolean isShallow()
                         .collect(Collectors.toSet());
     }
 
+    @Override
+    public void setTags(final Map<String, String> tags)
+    {
+        if (tags != null)
+        {
+            this.tags = new HashMap<>(tags);
+        }
+        else
+        {
+            this.tags = new HashMap<>();
```

Comment:
@matthieun In the case that the parameter `tags` is `null`, should we be setting `this.tags` to an empty map? Or should we be doing `this.tags = null;`. Curious to hear your thoughts.

Reply:
I believe we should keep it null in that case, as shallow version contain null members.